### PR TITLE
feat(tbd-4445): Add support for EMR 5.5.0

### PR DESCRIPTION
**Pull request by rdubois**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550HCatalogModuleGroup.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package org.talend.hadoop.distribution.emr550.modulegroup;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.talend.hadoop.distribution.DistributionModuleGroup;
+
+public class EMR550HCatalogModuleGroup {
+
+    public static final String MODULE_GROUP_NAME = "HDFS-LIB-EMR_5_5_0_LATEST"; //$NON-NLS-1$
```
Comment (StefanLMNG) [inline main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550HCatalogModuleGroup.java]: Wrong value for the module group name ?

File: main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550SparkStreamingModuleGroup.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.talend.hadoop.distribution.DistributionModuleGroup;
+import org.talend.hadoop.distribution.condition.BasicExpression;
+import org.talend.hadoop.distribution.condition.ComponentCondition;
+import org.talend.hadoop.distribution.condition.EqualityOperator;
+import org.talend.hadoop.distribution.condition.SimpleComponentCondition;
+import org.talend.hadoop.distribution.constants.SparkStreamingConstant;
+
+public class EMR550SparkStreamingModuleGroup {
+
+    public static final String MODULE_GROUP_NAME = "SPARK2-LIB-EMR_5_5_0_LATEST"; //$NON-NLS-1$
```
Comment (StefanLMNG) [inline main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550SparkStreamingModuleGroup.java]: Same value for module group name like spark batch ?

File: main/plugins/org.talend.hadoop.distribution.emr500/plugin.xml [hunk 3]
```diff
@@ -1041,7 +1041,6 @@
 			 <library id="hk2-api-2.4.0-b34.jar"/>
 			 <library id="hk2-locator-2.4.0-b34.jar"/>
 			 <library id="hk2-utils-2.4.0-b34.jar"/>
-			 <library id="hive-common-emr500-latest"/>
 			 <library id="htrace-core-3.1.0-incubating.jar"/>
```
Comment (StefanLMNG) [inline main/plugins/org.talend.hadoop.distribution.emr500/plugin.xml]: Code formatting is not correct

File: main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550SparkStreamingModuleGroup.java (hunk 2 above)
Comment (rdubois) [inline main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550SparkStreamingModuleGroup.java]: Yes, it's needed for the spark.yarn.jars property.

File: main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550HCatalogModuleGroup.java (hunk 1 above)
Comment (rdubois) [inline main/plugins/org.talend.hadoop.distribution.emr550/src/main/java/org/talend/hadoop/distribution/emr550/modulegroup/EMR550HCatalogModuleGroup.java]: HCatalog requires HDFS and CXF dependencies. The CXF dependencies are directly imported into the component XML. Only the HDFS are missing. It's not a typo.

File: main/plugins/org.talend.hadoop.distribution.emr500/plugin.xml (hunk 3 above)
Comment (rdubois) [inline main/plugins/org.talend.hadoop.distribution.emr500/plugin.xml]: This plugin.xml is the file for EMR 5.0.0, not EMR 5.5.0. We should review it, but can be done out of this PR.

**Pull request merged**